contracts/pair/src/contract.rs

```diff
@@ -478,6 +483,22 @@ pub fn try_swap(
         ]))
 }
 
+pub fn try_update_configs(
+    deps: DepsMut,
+    info: MessageInfo,
+    end_time: Option<Option<u64>>,
+    commission_rate: Option<String>,
+) -> Result<Response, ContractError> {
+    let mut pair_info: PairInfo = PAIR_INFO.load(deps.storage)?;
+    if info.sender != pair_info.owner {
+        return Err(StdError::generic_err("Unauthorized"));
+    }
+    pair_info.end_time = end_time.unwrap_or(pair_info.end_time);
```

Comment:
Does it mean that `end_time` cannot be set as `None`?
I think we need to set the `end_time` as `None` post-LBP.
What do you think about it?

Reply:
No, It means that if the `end_date` is not provided, the `config.end_date` stays the same. To make the `end_date` == `None` it would be passed `Some(None)` where the first Option is unwrapped and then `None` stored.

The LBP pair logic is fine as is I think, since state_dates and end_dates are needed for weight calculation changes, so no need to modify that.